### PR TITLE
Option to not use Load Balancer with Fakeintake

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Add options FakeIntakeStackDef to be able to not use Load Balancer with the Fake Intake
Depends on: https://github.com/DataDog/test-infra-definitions/pull/403

### Motivation

Be able to not use the load balancer in some tests

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ require (
 	// `TEST_INFRA_DEFINITIONS_BUILDIMAGES` matches the commit sha in the module version
 	// Example: 	github.com/DataDog/test-infra-definitions v0.0.0-YYYYMMDDHHmmSS-0123456789AB
 	// => TEST_INFRA_DEFINITIONS_BUILDIMAGES: 0123456789AB
-	github.com/DataDog/test-infra-definitions v0.0.0-20231017144910-d2510f65b6fe
+	github.com/DataDog/test-infra-definitions v0.0.0-20231020114206-b6e2c8b2a497
```
Comment (pducolin) [inline test/new-e2e/go.mod:18]: **🔨 warning**
It's probably in the plan, adding here as a reminder to update `TEST_INFRA_DEFINITIONS_BUILDIMAGES`

**Review by pducolin — COMMENTED** [review]
Waiting for the PR in test-infra to be merged to approve this one, well done !

File: test/new-e2e/go.mod (hunk 1 above)
Comment (KevinFairise2) [reply test/new-e2e/go.mod]: Yes was in the plan, I updated both the module and the image with the latest commit from `test-infra-definitions` main branch

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Review by pducolin — APPROVED** [review]

**Pull request merged**